pkg/repository/v1/task.go

```diff
@@ -2843,21 +2923,19 @@ func (r *TaskRepositoryImpl) ReplayTasks(ctx context.Context, tenantId string, t
 			}
 		}
 
-		input, err := r.payloadStore.Retrieve(ctx, RetrievePayloadOpts{
+		retrieveOpt := RetrievePayloadOpts{
```

Comment:
@abelanger5 FYI, this was N+1 also - I just removed the singleton `Retrieve` method too